### DISCUSSION TITLE
fork到本地vscode的eslint和格式化与项目的对不上，ts很多类型不全

### DISCUSSION THREAD

**Discussion by kiss-keray** [opentiny/tiny-editor#80] [Q&A]
fork到本地想加一些功能太难了  点击保存就会格式化其他地方，eslint报错也多不好确定问题

**Comment by kiss-keray** [comment]
我最近在看custom-uploader没明白为什么要把图片和其他文件分开处理，图片也是一种文件。应该统一处理才对。我看video又当文件处理，在生成Delta时进行特殊处理了。这里是不是可以根据文件类型使用责任链默认来处理不同的文件类型？

**Reply by kagol** [reply]
> 我最近在看custom-uploader没明白为什么要把图片和其他文件分开处理，图片也是一种文件。应该统一处理才对。我看video又当文件处理，在生成Delta时进行特殊处理了。这里是不是可以根据文件类型使用责任链默认来处理不同的文件类型？

@kiss-keray 有想法可以一起优化下

**Reply by kiss-keray** [reply]
> > 我最近在看custom-uploader没明白为什么要把图片和其他文件分开处理，图片也是一种文件。应该统一处理才对。我看video又当文件处理，在生成Delta时进行特殊处理了。这里是不是可以根据文件类型使用责任链默认来处理不同的文件类型？
> 
> @kiss-keray 有想法可以一起优化下

https://github.com/slab/quill/pull/4423  我提了这个pr 我本地已经使用这种模式

**Comment by zzxming** [comment]
编辑器是不是开启了 prettier 格式化？当前 main 分支通过 eslint . 检查是没有错误的

**Reply by kiss-keray** [reply]
eslint我看是格式化后报错的，不格式报错的都是ts问题。格式化这个不是应该在.vscode里面配置好这些吗？隔壁的quill我在本地格式化这些与源码的格式化一致的就没问题。
<img width="817" alt="image" src="https://github.com/user-attachments/assets/298de1ad-d539-494e-a3f1-64ec9f12e562">

**Reply by zzxming** [reply]
ts报错问题是有的，项目最开始的时候几乎没考虑ts类型问题，我处理过的模块是都有ts类型的，但是还有很多是没有处理的

项目有prettier配置残留，如果你的问题不是因为prettier的话，希望能提供一下你的配置信息方便找出问题

**Reply by kiss-keray** [reply]
> ts报错问题是有的，项目最开始的时候几乎没考虑ts类型问题，我处理过的模块是都有ts类型的，但是还有很多是没有处理的
> 
> 项目有prettier配置残留，如果你的问题不是因为prettier的话，希望能提供一下你的配置信息方便找出问题

找到问题了，因为我的vscode默认设置配置了使用prettier格式化
```json
"[typescript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[vue]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[css]": {
    "editor.defaultFormatter": "michelemelluso.code-beautifier"
  },
  "[scss]": {
    "editor.defaultFormatter": "michelemelluso.code-beautifier"
  },
```
但是当前项目提交的都是使用的eslint格式化，我现在在项目的.vscode的setting.json配置默认eslint格式化就好了。这个配置应该提交了使得每个人获取代码后能按项目设置eslint格式化。
```json
  "[typescript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
  },
  "[javascript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
  }
```

**Reply by zzxming** [reply]
感谢你的反馈，我提交了修复的pr，合并后会在此项目禁用prettier

**Reply by kagol** [reply]
@kiss-keray PR #85 已合入，感谢 @zzxming 的贡献👍
